x-pack/plugin/old-lucene-versions/src/main/java/org/elasticsearch/xpack/lucene/bwc/codecs/BWCCodec.java

```diff
@@ -38,44 +31,22 @@
 import org.apache.lucene.index.Terms;
 import org.apache.lucene.store.Directory;
 import org.apache.lucene.store.IOContext;
+import org.elasticsearch.xpack.lucene.bwc.codecs.lucene70.BWCLucene70Codec;
 
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Iterator;
 import java.util.List;
 
-public class BWCLucene70Codec extends Codec {
-
-    private final FieldInfosFormat fieldInfosFormat;
-    private final SegmentInfoFormat segmentInfosFormat;
-    private final LiveDocsFormat liveDocsFormat;
-    private final CompoundFormat compoundFormat;
-    private final PostingsFormat postingsFormat;
-    private final StoredFieldsFormat storedFieldsFormat;
-
-    public BWCLucene70Codec() {
-        super("BWCLucene70Codec");
-        fieldInfosFormat = wrap(new Lucene60FieldInfosFormat());
-        segmentInfosFormat = wrap(new Lucene70SegmentInfoFormat());
-        liveDocsFormat = new Lucene50LiveDocsFormat();
-        compoundFormat = new Lucene50CompoundFormat();
-        postingsFormat = new EmptyPostingsFormat();
-        storedFieldsFormat = new Lucene50StoredFieldsFormat(Lucene50StoredFieldsFormat.Mode.BEST_SPEED);
-    }
-
-    @Override
-    public FieldInfosFormat fieldInfosFormat() {
-        return fieldInfosFormat;
-    }
+/**
+ * Base class for older BWC codecs
+ */
+public abstract class BWCCodec extends Codec {
 
-    @Override
-    public SegmentInfoFormat segmentInfoFormat() {
-        return segmentInfosFormat;
-    }
+    private final PostingsFormat postingsFormat = new EmptyPostingsFormat();
```

Comment:
Let's fix Lucene to only pull the postings format if any of the fields has postings, like we already do for points and doc values?